core/state/state_object.go

```diff
@@ -246,8 +223,13 @@ func (self *StateObject) Code() []byte {
 	return self.code
 }
 
+func (self *StateObject) CodeHash() common.Bytes {
+	return self.codeHash
+}
```

Comment:
Any particular reason for having this method? I only see a single use of this, where we might as well directly fetch the `codeHash` field. That would maybe be a bit cleaner as I originally assumed that this create a deep copy of the hash.